src/orange/OrangeTrackView.hh

```diff
@@ -217,20 +248,31 @@ OrangeTrackView::operator=(Initializer_t const& init)
 
         auto lsa = this->make_lsa(LevelId{level});
         lsa.vol() = tinit.volume;
-        lsa.pos() = init.pos;
-        lsa.dir() = init.dir;
+        lsa.pos() = local.pos;
+        lsa.dir() = local.dir;
         lsa.universe() = uid;
         lsa.surf() = LocalSurfaceId{};
         lsa.sense() = Sense{};
         lsa.boundary() = BoundaryResult::exiting;
 
-        auto const& vol_rec = tracker.unit_record().volumes[tinit.volume];
-        next_uid = params_.volume_records[vol_rec].daughter;
-        ++level;
+        next_uid = tracker.daughter_uid(tinit.volume);
+
+        if (next_uid)
+        {
+            auto const& trans = tracker.translation(tinit.volume);
```

Comment:
I was thinking along the lines of:
```suggestion
        daughter_id = tracker.daughter(tinit.volume);

        if (daughter_id)
        {
            TranslationId trans_id = params_.daughters[daughter_id];
            auto const& trans = params_.translations[trans_id];
```
since we might end up repeating the same code for the other universe types. Thoughts?

Reply:
I agree that this is better. The only downside is it locks us into daughters being stored the same way for array units, which they probably will be, but I haven't thought about it enough. I will change to this.